src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -277,13 +284,47 @@ internal void PublishRpcChannelReadyEvent(RpcEvent initEvent)
                 return;
             }
 
-            RpcChannelReadyEvent readyEvent = new RpcChannelReadyEvent(_workerId, _workerConfig.Language, this, _initMessage.WorkerVersion, _initMessage.Capabilities);
-            _eventManager.Publish(readyEvent);
+            var managedDepedencyProvider = new ManagedDependencyProvider();
+            var managedDependencies = managedDepedencyProvider.GetDependencies(Path.Combine(_rootScriptPath, ScriptConstants.HostMetadataFileName), SystemEnvironment.Instance, _managedDependencyOptions);
+            string managedDependencyCapability;
+            if (_initMessage.Capabilities.TryGetValue(EnumDependencies.DependenciesInstalltionEnabled.ToString(), out managedDependencyCapability)
+                && string.Equals(managedDependencyCapability, EnumDependencies.True.ToString(), StringComparison.InvariantCultureIgnoreCase)
+                && managedDependencies?.Any() == true)
+            {
+                _inboundWorkerEvents.Where(msg => msg.MessageType == MsgType.DependencyDownloadResponse)
+                    .Subscribe(PublishRpcChannelReadyEvent, HandleManagedDependencyError);
 
-            if (_functionRegistrations != null)
+                //Send the message to download dependencies
+                _workerChannelLogger?.LogInformation($"Sending dependency download request to {_workerConfig.Language} language worker");
+                DownloadDependencyRequest downloadDependencyRequest = new DownloadDependencyRequest();
+                downloadDependencyRequest.Dependencies.AddRange(managedDependencies);
+
+                SendStreamingMessage(new StreamingMessage
+                {
+                    DownloadDependencyRequest = downloadDependencyRequest
+                });
+
+                _workerChannelLogger?.LogInformation($"Successfully sent dependency download request to {_workerConfig.Language} language worker");
+            }
+            else
             {
-                RegisterFunctions(_functionRegistrations);
+                PublishRpcChannelReadyAndRegisterFunction();
+            }
+        }
+
+        internal void PublishRpcChannelReadyEvent(RpcEvent dependencyDownloadEvent)
+        {
+            _startLatencyMetric?.Dispose();
+            _startLatencyMetric = null;
+
+            var managedDependencyResponse = dependencyDownloadEvent.Message.DependencyDownloadResponse;
```

Comment:
Same here, this needs to be decoupled. Publishing channel ready event should not be handling managed dependency error